com.unity.sg2/Editor/GraphUI/DataModel/ShaderGraphModel.cs

```diff
@@ -31,93 +32,40 @@ class MainPreviewData
         [NonSerialized]
         public float scale = 1f;
 
-        public void Initialize()
+        public Mesh mesh
         {
-            if (serializedMesh.IsNotInitialized)
-            {
-                // Initialize the sphere mesh as the default
-                Mesh sphereMesh = Resources.GetBuiltinResource(typeof(Mesh), $"Sphere.fbx") as Mesh;
-                serializedMesh.mesh = sphereMesh;
-            }
+            get => serializedMesh.mesh;
+            set => serializedMesh.mesh = value;
         }
     }
 
     public class ShaderGraphModel : GraphModel
     {
-        public GraphHandler GraphHandler => ShaderGraphAssetModel.GraphHandler;
-
-        public ShaderGraphAssetModel ShaderGraphAssetModel => Asset as ShaderGraphAssetModel;
-
-        public Registry RegistryInstance => ((ShaderGraphStencil)Stencil).GetRegistry();
-
-        #region MainPreviewData
-        MainPreviewData m_MainPreviewData = new ();
-
-        internal MainPreviewData mainPreviewData => m_MainPreviewData;
-
-        public void SetPreviewMesh(Mesh newPreviewMesh)
+        [SerializeField]
+        private SerializableGraphHandler graphHandlerBox = new();
+        [SerializeField]
+        private SerializableTargetSettings targetSettingsBox = new();
+        [SerializeField]
+        private MainPreviewData mainPreviewData = new();
+        [SerializeField]
+        private bool isSubGraph = false;
+
+        internal GraphHandler GraphHandler => graphHandlerBox.Graph;
+        internal Registry RegistryInstance => graphHandlerBox.Graph.registry;
+        internal List<JsonData<Target>> Targets => targetSettingsBox.Targets;
+        internal MainPreviewData MainPreviewData => mainPreviewData;
+        internal bool IsSubGraph => CanBeSubgraph();
+
+        public void Init(GraphHandler graph, bool isSubGraph)
         {
-            m_MainPreviewData.serializedMesh.mesh = newPreviewMesh;
-        }
-
-        public void SetPreviewScale(float newPreviewScale)
-        {
-            m_MainPreviewData.scale = newPreviewScale;
-        }
-
-        public void SetPreviewRotation(Quaternion newRotation)
-        {
-            m_MainPreviewData.rotation = newRotation;
-        }
-
-        public void SetPreviewSize(Vector2 newPreviewSize)
-        {
-            m_MainPreviewData.width = Mathf.FloorToInt(newPreviewSize.x);
-            m_MainPreviewData.height = Mathf.FloorToInt(newPreviewSize.y);
-        }
-
-        public void SetPreviewRotationLocked(bool preventRotation)
-        {
-            m_MainPreviewData.preventRotation = preventRotation;
-        }
-
-        #endregion
```

Comment:
I removed these modifiers to mainPreviewData and instead just made it a public. These weren't in use anyway and the MainPreviewData object (defined above) can own its own interface.